sdk/nodejs/resource.ts

```diff
@@ -149,6 +186,28 @@ export abstract class Resource {
                 opts.protect = opts.parent.__protect;
             }
 
+            for (const parentAlias of opts.parent.__aliases) {
+                if (!opts.aliases) {
+                    opts.aliases = [];
+                }
+
+                let aliasName = output(name);
+                // If the child name has the parent name as a prefix, then we make the assumption that it was
+                // constructed from the convention of using `{name}-details` as the name of the child resource.  To
+                // ensure this is aliased correctly, we must then also replace the parent aliases name in the prefix of
+                // the child resource name.
+                if (name.startsWith(opts.parent.__name)) {
+                    const parentName = opts.parent.__name;
+                    aliasName = output(parentAlias).apply(parentAliasUrn => {
+                        const parentAliasName = parentAliasUrn.substring(parentAliasUrn.lastIndexOf("::") + 2);
+                        return parentAliasName + name.substring(parentName.length);
+                    });
+                }
+
+                const childAlias = createUrn(aliasName, t, parentAlias);
+                opts.aliases.push(childAlias);
+            }
```

Comment:
1. any reason we can't move all this work done below to where we do the other alias work?

Reply:
This is quite different work - related to the existence of a parent and the inheritance of implicit resource options implied by having a parent.  Doing this in the same place as all the other parent inheritance seems to make sense.